### PR TITLE
Improvements to 2.5 migration note and readme

### PR DESCRIPTION
**Pull request by emanic**
## Description
- Edits text of note on Releases page
- Changes styling of note to be more prominent and noticeable
- Edits text of readme.md for clarity
- Updates YML to alleviate warning msg when building site

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by emanic** [comment]
The styling of the note will be red as shown here https://getbootstrap.com/docs/3.3/components/#alerts

File: upgrade/v2.5/README.md [hunk 1]
```diff
@@ -60,31 +55,29 @@ Since we will need `calicoctl` versions [v1.4.1](https://github.com/projectcalic
...
+  2.2. Check the job status `kubectl describe job/calico-upgrade-v2.5`.
 
-## 3. After the migration
+  2.3. Check the upgrade job logs `kubectl logs <upgrade-pod-name>` to make sure there aren't any errors. (You can get the upgrade job's pod name from the previous command output `Message` field.)
 
-### 3.1 Verify that the data has been copied correctly
+## 3. Verify that the data has been copied correctly
 
-Make sure you have all the `IPPools`, `BGPPeers`, `GlobalFelixConfig` and `GlobalBGPConfig` resource configs you had before the upgrade:
+Make sure you have all the `IPPools`, `BGPPeers`, `GlobalFelixConfig`, and `GlobalBGPConfig` resource configs you had before the upgrade:
```
Comment (gunjan5) [inline upgrade/v2.5/README.md]: `before the upgrade` -> `before the migration`

File: upgrade/v2.5/README.md [hunk 2]
```diff
@@ -1,52 +1,47 @@
-# Calico v2.5.0 Upgrade Procedure
+# Migrating Kubernetes datastore data before upgrading to Calico v2.5.0 or later
```
Comment (gunjan5) [inline upgrade/v2.5/README.md]: This doc is for both migration and updating calico version, not just for migration. I think we can just say `Upgrading Calico to v2.5.0+ with Kubernetes datastore backend` or something similar? WDYT?

File: Gemfile.lock [hunk 3]
```diff
@@ -0,0 +1,216 @@
+GEM
```
Comment (gunjan5) [inline Gemfile.lock]: what are these for?

File: upgrade/v2.5/README.md (hunk 2 above)
Comment (emanic) [inline upgrade/v2.5/README.md]: Oh, OK, I thought it was just for migration. Let me fix this then.

File: Gemfile.lock (hunk 3 above)
Comment (emanic) [inline Gemfile.lock]: Yes, I think the .lock file is unnecessary. I will remove it. The Gemfile is something that will make it easier for people to build the site...I can take it out of this commit if it is bothersome. It's not directly related to this PR.

Comment (gunjan5) [inline Gemfile.lock]: > The Gemfile is something that will make it easier for people to build the site

if it's related to building the site then I'll leave it to @ozdanborne

File: _config.yml [hunk 4]
```diff
@@ -4,7 +4,7 @@
 # Make our URLs look nice. We're not a blog with dated posts, we just want the title.
 permalink:     /:title.html
 
-gems:
+plugins:
```
Comment (gunjan5) [inline _config.yml]: is this the reason why the build is failing?

Comment (bcreane) [inline _config.yml]: looks more like something wasn't right when I checked in the final v2.5.0 doc changes (e.g. moving "latest" link from 2.4 to 2.5).

Comment (bcreane) [inline _config.yml]: actually, @gunjan5 is right. @emanic - if you run `make htmlproofer` you'll see the broken links.

Comment (emanic) [inline _config.yml]: OK let me do this, I have not tried `htmlproofer` yet.

Comment (emanic) [inline _config.yml]: I changed it back to `gems` but `htmlproofer` still returns 2410 errors (same as with `plugins`), so I am not sure that this line is the reason for the issue.

**Comment by bcreane** [comment]
LGTM. We need a repo maintainer - @gunjan5 , @ozdanborne - to accept the PR.

**Comment by gunjan5** [comment]
@emanic can you squash the commits? thanks

**Comment by emanic** [comment]
OK sure @gunjan5

**Comment by emanic** [comment]
I have now managed to run `make htmlproofer` locally and found one broken link. Let me fix that before we merge @gunjan5

**Comment by emanic** [comment]
OK all fixed and squashed @gunjan5 👍

**Pull request merged**